src/main/java/com/aws/greengrass/deployment/DeploymentDocumentDownloader.java

```diff
@@ -192,9 +205,9 @@ private GetDeploymentConfigurationResponse getDeploymentConfiguration(String dep
         } catch (SdkClientException e) {
             throw new RetryableDeploymentDocumentDownloadException(
                     "Failed to contact Greengrass cloud or unable to parse response", e);
+        } catch (Exception e) {
+            throw new RuntimeException(e);
```

Comment:
why do you need this? this looks wrong

Reply:
Because `runWithRetry` throws `Exception`, may I have some suggestions on handling it?